switchmap/server/db/table/macport.py

```diff
@@ -134,7 +134,12 @@ def update_row(idx, row):
         None
 
     """
-    # Update
+    # checking for duplication
+    existing = exists(row.idx_l1interface, row.idx_mac)
+    if existing and existing.idx_macport != idx:
```

Comment:
Revert. There is no way to determine whether the update is a success. This file must remain unchanged.

Reply:
This is also an invalid test. You are comparing two different object types and the update will never happen.